aspnetcore/blazor/components/css-isolation.md

```diff
@@ -0,0 +1,148 @@
+
+## CSS isolation bundling
+
+CSS isolation occurs at build time. During this process, Blazor rewrites CSS selectors to only match markup rendered by the component. These rewritten CSS styles are bundled together and produced as a static web asset at `_framework/scoped.styles.css`.
+
+In the app, reference the bundled file by inspecting the reference inside the `<head>` tag of the generated HTML:
+
+```html
+<link href="_framework/scoped.styles.css" rel="stylesheet">
+```
+
+The bundle associates each component with a scope identifier. For each styled component, an HTML attribute is appended with the format `b-<10-character-string>`. For example, in the rendered `Counter` component, Blazor appends a scope identifier to the `h1` (this is unique and different for each app):
+
+```html
+<h1 b-3xxtam6d07>
+```
+
+The `scoped.styles.css` file uses the scope identifier to group a style declaration with its component.
+
+```css
+/* /Pages/Counter.razor.rz.scp.css */
+h1[b-3xxtam6d07] {
+    color: brown;
+    font-family: Tahoma, Geneva, Verdana, sans-serif;
+}
+```
+
+Additionally, at build time a project bundle is created with the convention `StaticWebAssetsBasePath/MyProject.lib.scp.css`. This is referenced when projects are used for NuGet packages or a Razor Class Library, which both support CSS isolation. This `lib.scp.css` file is not published as a static web asset.
+
+## CSS preprocessor support
+
+CSS preprocessors are useful with improving CSS development by utilizing features like variables, nesting, modules, mixins, and inheritance. While CSS isolation does not *natively* support CSS preprocessors such as Sass or Less, integrating them is seamless as long as preprocessor compilation occurs before Blazor rewrites the CSS selectors during the build process.
+
+For example, in the Visual Studio Task Runner Explorer, configure existing preprocessor compilation as a **Before Build** task.
+
+Many third-party NuGet packages, such as [Delegate.SassBuilder](https://www.nuget.org/packages/Delegate.SassBuilder/), can compile SASS/SCSS files at the beginning of the build process before CSS isolation occurs—preventing any additional configuration.
+
+For example, when you add a `MyComponent.razor.scss` file, these tools can compile SASS changes to `MyComponent.razor.css` at the beginning of the build process. With that step complete, the isolation step is ready. 
+
+## CSS isolation configuration
+
+When working with isolated CSS, no configuration is required. However, configuration is available for advanced scenarios or when there are dependencies on other existing tools or workflows.
+
+### Customize scope identifier format
+
+By default, scope identifiers use the format `b-<10-character-string>`. To customize the scope identifier format, update the project file to a desired pattern.
+
+```xml
+<ItemGroup>
+    <None Update="MyComponent.razor.css" CopyToOutputDirectory="my-custom-scope-identifier" />
```

Comment:
`CopyToOutputDirectory` this seems wrong.

Reply:
@javiercn - definitely is. I replaced `CopyToOutputDirectory` to `CssScoped` in my latest push. Apologies for the brain fart (probably muscle memory from years of `CopyToOutputDirectory`).